lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -2179,6 +2179,8 @@ def main():
             tmp_result = set_vm_power_state(pyv.content, vm, module.params['state'], module.params['force'], module.params['state_change_timeout'])
             if tmp_result['changed']:
                 result["changed"] = True
+                if module.params['wait_for_ip_address']:
+                    pyv.wait_for_vm_ip(vm)
```

Comment:
I think we should add an additional check to check for this parameter only if powerstate is `poweredon`  or `rebootguest` otherwise in other options module will wait for IP address.